classes/class-p4-master-site.php

```diff
@@ -674,7 +674,9 @@ public function p4_auto_generate_excerpt( $post_id, $post ) {
 			remove_action( 'save_post', [ $this, 'p4_auto_generate_excerpt' ], 10 );
 
 			// Generate excerpt text.
-			$post_excerpt   = strip_shortcodes( $post->post_content );
+			preg_match( '<p>(.*?)</p>', $post->post_content, $match_paragraph );
+
+			$post_excerpt   = count( $match_paragraph ) > 0 ? $match_paragraph[1] : strip_shortcodes( $post->post_content );
```

Comment:
Are we sure there can be no shortcodes inside the first paragraph? If there could be then you could put this ternary inside the `strip_shortcodes` call.

Reply:
Good point.